inst/private/doctest_collect.m

```diff
@@ -283,6 +291,34 @@
 end
 
 
+function targets = collect_targets_octfile (file, depth)
+  % first target is the name of the octfile (w/o extension)
+  [~, basename, ext] = fileparts (file);
+  assert (strcmp (ext, '.oct'))
+  target = collect_targets_function (basename);
+  target.name = file;
+  target.depth = depth;
+  targets = [target];
+
+  % octfile may have many fcns in it: find them using the autoload map
+  A = fullfile (pwd, file);
```

Comment:
This only works in the limited case where `foo.oct` is in the current working directory. IOW, exactly when it was called from `doctest somedir` and `doctest` already did a `chdir`. What if we use `which` or `file_in_loadpath` to get the full file name of `[basename '.oct']`?

Reply:
Should we just use the match-only-the-filename approach from your PR?  Maybe that's good enough for now.  Do you think many people have a generic `utils.oct` or similar lying around if different directories?

There is some overlap here with the discussion in #87 about classes and whether only "locally" defined methods should be tested or *all* methods (such as those from superclasses defined elsewhere).

Reply:
It's probably good enough for a start if you'd rather stay simple and just use the basename. I really don't think non-developers have their own personal .oct files floating around. The much more likely case of the same .oct file is multiple copies in different directories, maybe with a `pkg load` and with a manual `addpath`, which should be ok.

I searched the entire Debian archive for any files ending with .oct, and I found 226 unique file names, no duplicates, which doesn't say anything about third-party or personal projects, but at least no well-known projects are using names like `common.oct` or `main.oct`.